hypertrace-graphql-entity-schema/src/main/java/org/hypertrace/graphql/entity/dao/IncomingEdgesDataFetcher.java

```diff
@@ -0,0 +1,54 @@
+package org.hypertrace.graphql.entity.dao;
+
+import graphql.schema.DataFetcher;
+import graphql.schema.DataFetchingEnvironment;
+import java.util.Collections;
+import java.util.Map;
+import java.util.Optional;
+import java.util.concurrent.CompletableFuture;
+import javax.inject.Inject;
+import org.hypertrace.core.graphql.common.fetcher.InjectableDataFetcher;
+import org.hypertrace.core.graphql.deserialization.ArgumentDeserializer;
+import org.hypertrace.graphql.entity.schema.EdgeResultSet;
+import org.hypertrace.graphql.entity.schema.Entity;
+import org.hypertrace.graphql.entity.schema.EntityType;
+import org.hypertrace.graphql.entity.schema.argument.NeighborEntityScopeArgument;
+import org.hypertrace.graphql.entity.schema.argument.NeighborEntityTypeArgument;
+
+public class IncomingEdgesDataFetcher extends InjectableDataFetcher<EdgeResultSet> {
```

Comment:
nit - you could probably share 95% of these two data fetchers. Line 54 is the only distinction, right?